addons/html_builder/static/src/core/save_snippet_plugin.js

```diff
@@ -38,7 +49,14 @@ export class SaveSnippetPlugin extends Plugin {
 
     async saveSnippet(el) {
         const cleanForSaveHandlers = this.getResource("clean_for_save_handlers");
-        const savedName = await this.config.saveSnippet(el, cleanForSaveHandlers);
+        const onWillSaveHandlers = this.getResource("on_will_save_snippet_handlers");
+        const onSavedHandlers = this.getResource("on_saved_snippet_handlers");
+        const savedName = await this.config.saveSnippet(
+            el,
+            cleanForSaveHandlers,
+            onWillSaveHandlers,
+            onSavedHandlers
+        );
```

Comment:
I prefer a mix between this solution and the last one. (Like ignoreDOMMutation). With this version, if you have a crash between onWillSaveHandlers and onSavedHandlers. You will not restart interactions